### DISCUSSION TITLE
Embed to express app

### DISCUSSION THREAD

**Discussion by antirek** [evidence-dev/evidence#2351] [General]
**Feature Description**
Embed evidence markdown view and render to express app.

**Goal of Feature**
Fast development complex app

**Comment by archiewood** [comment]
Hey @antirek,

We've been talking about creating a better embedded experience for Evidence, so it's great to hear that it's something you're trying to do. Can you give me a better idea of what it is you're trying to achieve:
1. **Embed specific components** from an Evidence project into your express app (eg Charts, Tables, Text snippets etc).
2. **Compile "Evidence flavoured markdown" pages into HTML** that you can insert as pages in your express app.
3. **Something else?**

If the answer is 2, then there may be a way to achieve that today:

Evidence uses SvelteKit, and compiles "Evidence Markdown" in the following way
1. SQL queries are extracted and run against your database, and stored in js objects.
2. Markdown is rendered into HTML pages. Any components are populated with data from the js objects.
3. If you run `npm run build` then static HTML pages are generated. Though I haven't done it myself I believe these could be [served by an express app](https://stackoverflow.com/questions/71983776/how-do-i-connect-an-expressjs-backend-to-a-sveltekit-frontend)

If the answer is 1, then we don't have a way to achieve that well today, but it's definitely on our to-do list!

**Comment by jcalfee** [comment]
Just curious, does the compiled markdown have a javascript entry point where it could do something seamless like render to a "div"?  It should simply a matter of importing some javascript and passing it a div id and a some configuration.  This is probably going to give the most functionality for the least amount of work and encourage custom evidence / markdown plugins when people have more customization needs so you may attract some more contributions there.
